torch/csrc/jit/testing/file_check.cpp

```diff
@@ -110,21 +114,64 @@ size_t assertFind(
   return pos;
 }
 
+size_t assertFindRegex(
+    const SourceRange& search_range,
+    const std::string& sub,
+    const std::function<void(std::ostream& out)>& extra_msg = nullptr) {
+  auto pos =
+      search_range.source()->text_str().find_regex(sub, search_range.start());
+
+  if (pos == std::string::npos) {
+    std::stringstream ss;
+    ss << "Expected to find regex ";
+    c10::printQuotedString(ss, sub);
+    ss << " but did not find it" << std::endl;
+    ss << "Searched string:" << std::endl;
+    if (extra_msg) {
+      extra_msg(ss);
+    }
+    throw std::runtime_error(ss.str());
+
+    return std::string::npos;
+  }
+  return pos;
+}
+
 size_t assertFind(
     const SourceRange& search_range,
     const std::string& sub,
-    const Check& check) {
-  return assertFind(search_range, sub, [&](std::ostream& out) {
-    out << "From " << check << "\n";
-  });
+    const Check& check,
+    const bool regex = false) {
+  return regex
+      ? (assertFindRegex(
+            search_range,
+            sub,
+            [&](std::ostream& out) { out << "From " << check << "\n"; }))
+      : (assertFind(search_range, sub, [&](std::ostream& out) {
+          out << "From " << check << "\n";
+        }));
 }
 
 size_t assertFind(
+    const std::shared_ptr<Source>& source,
+    const std::string& sub,
+    size_t start,
+    const Check& check,
+    const bool regex = false) {
+  return assertFind(
+      SourceRange(source, start, source->size()), sub, check, regex);
+}
+
+size_t assertFindRegex(
     const std::shared_ptr<Source>& source,
     const std::string& sub,
     size_t start,
     const Check& check) {
-  return assertFind(SourceRange(source, start, source->size()), sub, check);
+  return assertFind(
+      SourceRange(source, start, source->size()),
+      sub,
+      check,
+      true /* regex match */);
```

Comment:
I feel that the benefit of sharing code with assertFind is less than the cost of readability. It might be better to just have a standalone path for assertFindRegex without modifying assertFind.